source/types.ts

```diff
@@ -137,6 +137,13 @@ export interface Options<T> {
 	*/
 	clearInvalidConfig?: boolean;
 
+	/**
+	The mode that will be used for the config file.
+
```

Comment:
The docs should explain when it would be useful. Most devs would not need this, and from experience, setting `mode` can cause a lot of problems. For example, user run a tool with `sudo` and then later not with `sudo` and then the config file is not readable, etc. I did the mode thing in a different config library, and it caused endless of support requests.

For example, on macOS, `conf` by default, stores the config in `~/Library/Preferences`, which is not world writable. So this would not be needed if you target only macOS.

https://github.com/sindresorhus/env-paths#pathsconfig